sdcm/cluster.py

```diff
@@ -322,15 +323,103 @@ def __init__(self, name, ssh_login_info=None, base_logdir=None, node_prefix=None
         self.is_enterprise = None
         self.replacement_node_ip = None  # if node is a replacement for a dead node, store dead node private ip here
         self._set_prometheus_paths()
+        self.distro = None
+
+    def _get_distro(self):
+        # Distro attribute won't be changed, only need to detect once.
+        if self.distro:
+            return self.distro
+
+        result = self.remoter.run('cat /etc/redhat-release', ignore_status=True)
+        if 'CentOS' in result.stdout and 'release 7.' in result.stdout:
+            self.distro = Distro.CENTOS7
+        if 'Red Hat Enterprise Linux' in result.stdout and 'release 7.' in result.stdout:
+            self.distro = Distro.RHEL7
+
+        result = self.remoter.run('cat /etc/issue', ignore_status=True)
+        if 'Ubuntu 14.04' in result.stdout:
+            self.distro = Distro.UBUNTU14
+        elif 'Ubuntu 16.04' in self.stdout:
+            self.distro = Distro.UBUNTU16
+        elif 'Debian GNU/Linux 8' in result.stdout:
+            self.distro = Distro.DEBIAN8
+        elif 'Debian GNU/Linux 9' in result.stdout:
+            self.distro = Distro.DEBIAN9
+        else:
+            self.log.debug("Failed to detect the distro name, %s" % result.stdout)
+
+        return self.distro
+
+    def is_centos7(self):
+        self.distro == Distro.CENTOS7
+
+    def is_rhel7(self):
+        self.distro == Distro.RHEL7
+
+    def like_centos(self):
```

Comment:
It is better to say `is_redhat_like` since CentOS is open source distribution that is RHEL compatible

Reply:
I will use `is_rhel_like`